src/core/window.extern.js

```diff
@@ -22,5 +22,141 @@
 /** @type {undefined} */
 window.__AMP_ASSERTION_CHECK;
 
-/** @type {undefined|function(this:Window,!Error,Element=)} */
+/** @type {boolean|undefined} */
+window.IS_AMP_ALT;
+
+/** @type {boolean|undefined} */
+window.AMP_DEV_MODE;
+
+/** @type {Object} */
+window.__AMP_MODE = {};
+window.__AMP_MODE.a4aId;
+window.__AMP_MODE.development;
+window.__AMP_MODE.esm;
+window.__AMP_MODE.examiner;
+window.__AMP_MODE.lite;
+window.__AMP_MODE.localDev;
+window.__AMP_MODE.log;
+window.__AMP_MODE.minified;
+window.__AMP_MODE.rtvVersion;
+window.__AMP_MODE.runtime;
+window.__AMP_MODE.test;
+window.__AMP_MODE.version;
+
+/** @type {Object} */
+window.__AMP_LOG = {};
+window.__AMP_LOG.user;
+window.__AMP_LOG.dev;
+window.__AMP_LOG.userForEmbed;
```

Comment:
log is not in core yet, so I don't think its externs should be either. That aside, when log is added as an extern, I'd suggest we declare `window.__AMP_LOG` to be of type Log, which should be defined independently (like in log.extern.js possibly)